src/status/statusChanger.service.ts

```diff
@@ -0,0 +1,48 @@
+import { Inject, Injectable } from "@nestjs/common";
+import { Order } from "src/order/order.entity";
+import { Providers } from "src/providers";
+import { Repository } from "typeorm";
+import { OrderStatus } from "./orderStatus.entity";
+
+@Injectable()
+export class StatusChangerService {
+    private readonly MIN_DELAY = 10;
+    private readonly MAX_DELAY = 30;
+
+    constructor(
+        @Inject(Providers.StatusRepository)
+        private readonly statusRepository: Repository<OrderStatus>,
+        @Inject(Providers.OrderRepository)
+        private readonly orderRepository: Repository<Order>,
+    ) { }
+
+    registerOrder(uuid: string) {
+        console.debug(`Registered order ${uuid}`);
+        this.addStatuses(uuid);
+    }
+
+    async addStatuses(uuid: string) {
+        const order = await this.orderRepository.findOne({ where: { uuid: uuid } });
+        for (const status of StatusDetails) {
+            await this.statusRepository.save({ order: order, status: status.status, description: status.details });
```

Comment:
is it possible to rename some variables. `status: status.status` is not very meaningful